x/orderbook/msg_test.go

```diff
@@ -0,0 +1,224 @@
+package orderbook
+
+import (
+	"testing"
+
+	"github.com/iov-one/weave"
+	coin "github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/weavetest"
+)
+
+func TestValidateCreateOrderBookMsg(t *testing.T) {
```

Comment:
It seems like for every validation spec there are tests that covers if single field is missing or invalid. Maybe I'm saying maybe there is a golang tool that creates this test automatically. If there is not we can create at some point by ourselves for the sake of preventing developer mistake

Reply:
Good idea. This would be good to auto-test if possible.

One issue here, is for example the fact that `askTicker < bidTicker` is custom logic that needs to be manually tested. And some fields are required while others are not.

But some auto-gen test saying, "these fields are required",  "all fields must be valid if present" and covering those test... that would be great. There are some libraries like https://godoc.org/gopkg.in/go-playground/validator.v9 that can generate validation code given json tags. I am not sure about anything that auto-gen test code, but nice idea.